src/types.ts

```diff
@@ -1,5 +1,75 @@
-import { FunctionInput } from "@azure/functions";
+import {
+    FunctionInput,
+    FunctionHandler,
+    InvocationContext,
+    FunctionResult,
+    FunctionOutput,
+    FunctionTrigger,
+} from "@azure/functions";
+import { DurableOrchestrationClient, EntityState } from "../src/classes";
+import { DurableEntityBindingInfo } from "../src/durableentitybindinginfo";
+import { IEntityFunctionContext } from "../src/ientityfunctioncontext";
+import { IOrchestrationFunctionContext } from "../src/iorchestrationfunctioncontext";
+import { OrchestratorState } from "../src/orchestratorstate";
+import { DurableOrchestrationInput } from "../src/testingUtils";
 
+// orchestrations
+export type OrchestrationHandler = (
+    context: IOrchestrationFunctionContext
+) => Generator<unknown, unknown, any>;
+
+export type OrchestrationFunction = FunctionHandler &
+    ((
+        context: IOrchestrationFunctionContext,
+        orchestrationTrigger: DurableOrchestrationInput
+    ) => Promise<OrchestratorState>);
+
+export interface OrchestrationTrigger extends FunctionTrigger {
+    type: "orchestrationTrigger";
+}
+
+// entities
+export type EntityHandler<T> = (context: IEntityFunctionContext<T>) => void;
+
+export type EntityFunction<T> = FunctionHandler &
+    ((
+        context: IEntityFunctionContext<T>,
+        entityTrigger: DurableEntityBindingInfo
+    ) => Promise<EntityState>);
+
+export interface EntityTrigger extends FunctionTrigger {
+    type: "entityTrigger";
+}
+
+// activities
+export type ActivityHandler<T> = FunctionHandler &
+    ((context: InvocationContext, input: T) => FunctionResult<any>);
+
+export type ActivityOptions<T> = {
+    handler: ActivityHandler<T>;
+    extraInputs?: FunctionInput[];
+    extraOutputs?: FunctionOutput[];
+};
+
+export interface ActivityTrigger extends FunctionTrigger {
+    type: "activityTrigger";
+}
+
+// clients
 export interface DurableClientInput extends FunctionInput {
     type: "orchestrationClient";
 }
+
+export type DurableClientHandler = (
+    context: InvocationContext,
+    triggerInput: any,
+    orchestrationClient: DurableOrchestrationClient
+) => FunctionResult<any>;
```

Comment:
This is only required because we have a `df.client` convenience API, right? From what I gather, these `-Handler` types usually correspond to Triggers ("the handler is called by the trigger"), so I'm confused to see a handler for the DF client which is an input Binding.

Reply:
> This is only required because we have a `df.client` convenience API, right? 

Yes, and since we decided to remove the `df.client()` helper for this PR, I have also removed this type